src/riscv/lib/src/pvm/linux.rs

```
@@ -1009,6 +1013,101 @@ impl<M: ManagerBase> SupervisorState<M> {
+            parameters::Rlimit::Memlock => MemorySize::<MC>::RLIMIT_MEMLOCK,
+            parameters::Rlimit::As => MemorySize::<MC>::RLIMIT_AS,
+            parameters::Rlimit::Locks => parameters::RLIMIT_LOCKS,
+            parameters::Rlimit::Sigpending => parameters::RLIMIT_SIGPENDING,
+            parameters::Rlimit::Msgqueue => parameters::RLIMIT_MSGQUEUE,
+            parameters::Rlimit::Nice => parameters::RLIMIT_NICE,
+            parameters::Rlimit::Rtprio => parameters::RLIMIT_RTPRIO,
+            parameters::Rlimit::Rttime => parameters::RLIMIT_RTTIME,
+        };
+
+        core.main_memory.write(addr.to_machine_address(), limit)?;
+        core.main_memory.write(
+            (addr + size_of_val(&limit) as u64).to_machine_address(),
+            limit,
+        )?;
+        Ok(())
+    }
+
+    /// Handle `prlimit64` system call.
+    ///
+    /// See: <https://www.man7.org/linux/man-pages/man2/prlimit64.2.html>
+    fn handle_prlimit64<MC: MemoryConfig>(
+        &self,
+        core: &mut MachineCoreState<MC, M>,
+        _pid: parameters::ProcessId,
+        resource: parameters::Rlimit,
+        new_limit: VirtAddr,
+        old_limit: VirtAddr,
+    ) -> Result<u64, Error>
+    where
+        M: ManagerReadWrite,
+    {
+        // In jstz this is only used to set the maximum size of the process stack to infinity. In
+        // other programs, it can be used to set limits for system resources. Allowing programs to
+        // set these values to extreme values would inflate proof size, so is undesirable.
+        // Futhermore, a privileged process (such as the supervisor implemented in this PVM) has
+        // the ability to set these limits to arbitrary values.
+        // So it does! By ignoring this system call, and using its own predefined limits.
+
+        // If new_limit is not NULL, the system call is being used to write new limits, so ignore
+        // it
+        if new_limit != 0 {
+            return Ok(0);
+        }
```

Instead of silently ignoring it, we should probably fail here to avoid entering an inconsistent state. Perhaps you can return `EPERM`?

---

>The calling process did not have permission set limits for the process specified by pid.

Sure, no problem. [Done](https://github.com/tezos/riscv-pvm/compare/8d2bd931cc69b3f764ffed9995a2097c5c222662..562da1975bb4d5e68f80708247922f9c7ff56912#diff-c0656d3081a3cd2dcda31be427ed813c730aa6af7b274a0389534c85d609e3e7R1065).